client/vm.go

```diff
@@ -549,8 +550,26 @@ func (c *Client) waitForVmState(id string, stateConf StateChangeConf) error {
 	return err
 }
 
-func (c *Client) waitForModifyVm(id string, waitForIp bool, timeout time.Duration) error {
-	if !waitForIp {
+func (c *Client) waitForModifyVm(id string, waitForIp bool, bootAfterCreate bool, timeout time.Duration) error {
+	if !bootAfterCreate {
+		refreshFn := func() (result interface{}, state string, err error) {
+			vm, err := c.GetVm(Vm{Id: id})
+
+			if err != nil {
+				return vm, "", err
+			}
+
+			return vm, vm.PowerState, nil
+		}
+		stateConf := &StateChangeConf{
+			Pending: []string{"Running"},
+			Refresh: refreshFn,
+			Target:  []string{"Halted", "Stopped"},
+			Timeout: timeout,
+		}
+		_, err := stateConf.WaitForState()
+		return err
```

Comment:
This function is already a mess (which I'm guilty of :upside_down_face:), so I think in order to support managing `power_state` we need to leave this better than it currently is. My proposal for that was what I did in my partially implemented `power_state` change [here](https://github.com/terra-farm/terraform-provider-xenorchestra/compare/master...update-power_state-to-be-non-computed-vm-resource-argument#diff-27b07f0cb784dad37043e3062a4d86ed7ffc4029ac2e568eec2888b7db39b619R559-R570). 

Also from a quick glance, `Stopped` doesn't seem to be a real state ([source](https://github.com/vatesfr/xen-orchestra/blob/7c009b0fc0be09cb86045358b48cdf5a51d9f9b3/%40xen-orchestra/lite/src/libs/xen-api/xen-api.enums.ts#L67-L72)). So references to that should ideally be cleaned up. That is dependent on checking the XAPI source to verify that those are a complete set of power states.